app/js/arethusa.search/services/search.js

```diff
@@ -40,6 +40,182 @@ angular.module('arethusa.search').service('search', [
+           (match.matchedPrefix > maxPrefix && match.matchedSuffix == maxSuffix) ||
+           (match.matchedPrefix == maxPrefix && match.matchedSuffix > maxSuffix)) {
+            maxPrefix = match.matchedPrefix;
+            maxSuffix = match.matchedSuffix;
+            bestMatches = [match];
+          } else if (match.matchedPrefix == maxPrefix && match.matchedSuffix == maxSuffix) {
+            bestMatches.push(match)
+          }
+        });
+        bestMatches.forEach(function (match) {
+          memo.push(match.id);
+          if (match.includePrevious) {
+            memo.push(match.includePrevious);
+          }
+          if (match.includeNext) {
+            memo.push(match.includeNext);
+          }
+        })
+      });
+    };
+
+    /**
+     * compare two words, account for the fact that wordB may be represented by a combination
+     * of wordA with an enclytic that appears before or after it
+     * @param {String} wordA - token which may be a partial word
+     * @param {String} wordAPrev - token which appears before wordA (may be null)
+     * @param {String} wordANext - token which appears after wordA (may be null)
+     */
+    this.compareWordsWithContext = function(wordA,wordAPrev,wordANext,wordB) {
+      var match = compareWords(wordA,wordB);
+      var combine = 0;
+      // latin enclytics usually are preceded with a '-' and may be
+      // either right after the base word or shifted to right before it
+      if (!match && wordANext && wordANext.match(/^-/)) {
+        match = compareWords(wordA + wordANext.replace(/^-/,''),wordB);
+        if (match) {
+          combine = 1;
+        }
+      }
+      if (!match ) {
+        // greek krasis is postfixed with a - and should appear before the
+        // base word
+        if (wordAPrev && wordAPrev.match(/-$/)) {
+          match = compareWords(wordAPrev.replace(/-$/,'') + wordA, wordB);
+        } else if (wordAPrev && wordAPrev.match(/^-/)) {
+           // handles the case where the enclytic is shifted to before the word
+           match = compareWords(wordA + wordAPrev.replace(/^-/,''),wordB);
+        }
+        if (match) {
+          combine = -1;
+        }
+      }
+      // recheck to see if the word we're testing is the enclytic
+      if (!match && (wordA.match(/^-/) || wordA.match(/-$/))) {
+        var testWord;
+        if (wordA.match(/^-/)) {
+          wordA = wordA.replace(/^-/,'');
+        } else {
+          wordA = wordA.replace(/-$/,'');
+        }
+        if (wordAPrev) {
+          match = compareWords(wordAPrev + wordA,wordB);
+          if (!match) {
+            match = compareWords(wordA + wordAPrev,wordB);
+          }
+          if (match) {
+            combine = -1;
+          }
+        }
+        if (! match && wordANext) {
+          match = compareWords(wordA + wordANext,wordB);
+          if (!match) {
+            match = compareWords(wordANext + wordA,wordB);
```

Comment:
Not code related, but I'm also curious about this line and line 185 above. Are there cases where two words are considered equal but one has a proclitic and the other has no proclitic but an identical enclitic?

Reply:
I have no idea...